### PR TITLE
Fixed `hasInvalidEscape` implementation

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/55369
regression from https://github.com/microsoft/TypeScript/pull/51837 , cc @graphemecluster

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -5804,8 +5804,8 @@ function escapeTemplateSubstitution(str: string): string {
 /** @internal */
 export function hasInvalidEscape(template: TemplateLiteral): boolean {
     return template && !!(isNoSubstitutionTemplateLiteral(template)
-        ? template.templateFlags
-        : (template.head.templateFlags || some(template.templateSpans, span => !!span.literal.templateFlags)));
+        ? !!((template.templateFlags || 0) & TokenFlags.ContainsInvalidEscape)
```
Comment (jakebailey) [inline src/compiler/utilities.ts:5807]: This seems super noisy; can we pull this repeated bit out into a function?

Comment (Andarist) [reply src/compiler/utilities.ts]: ye, i had such a plan - pushed it out now 👍

**Comment by graphemecluster** [comment]
Oh, great that you spotted it – I overlooked that method.

**Comment by jakebailey** [comment]
I'm thinking this one deserves a backport; @DanielRosenwasser thoughts?

**Comment by jakebailey** [comment]
Hm, no, this was in 5.1; I misremembered which release we're currently on. Though, maybe it's worth doing anyway.

**Review by jakebailey — APPROVED** [review]

**Comment by jakebailey** [comment]
I'm also silly and didn't realize that the result of this bug is just "the generated code is worse", so, nevermind altogether then 😄

**Pull request merged**